query_plan_test.go

```diff
@@ -3,85 +3,198 @@ package main
 import (
 	"testing"
 
+	"github.com/google/go-cmp/cmp"
 	"google.golang.org/genproto/googleapis/spanner/v1"
+	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+func protojsonAsStruct(t *testing.T, j string) *structpb.Struct {
+	t.Helper()
+	var result structpb.Struct
+	if err := protojson.Unmarshal([]byte(j), &result); err != nil {
+		t.Fatal("protojsonAsStruct fails, invalid test case", err)
+	}
+	return &result
+}
+
+func TestRenderTreeWithStats(t *testing.T) {
+	for _, test := range []struct {
```

Comment:
Do we need to use table-driven tests even though there is a single test case?

Reply:
IMO, test cases should be added and refined repeatedly.
If it is not already table driven, there is some psychological hurdle to make table driven.

Reply:
Fair enough. Let's keep using current style.